doc/publishing.md

```diff
@@ -49,50 +51,135 @@ Each crate is "owned" by a list of maintainers, provided with the
+
+Note the path between `publish` and your non-commit revision. Likewise, you can
+run this command from outside your repository, as long as you supply the proper
+path to it.
+
+At this point, `alr publish` will carry out a few tests and, if everything
+checks out, it will create a `${repo_root}/alire/releases/crate-version.toml`
+file. This file must be submitted to the community index via a PR. A link for
+conveniently creating this PR will also be provided by `alr`:
+
+- Upload the generated index manifest file (`crate-version.toml`) to the
+  supplied page on github and create a pull-request.
+
+### Starting with a remote repository, without local clone
+
+This case is analogous to the previous one, but you don't need the local
+repository. The only difference is that you must supply the remote URL and
+commit (not a tag or branch):
+
+`alr publish <URL> <commit>`
+
+The checks will be carried out and the outcome will be the same. The remote
+repository still needs to be among the allowed ones.
+
+### Starting with a remote source archive
+
+This case can be used when you use another VCS other than `git`, or do not work
+with an online repository.
+
+In this use case, you start from an already prepared final remote tarball/zipball:
+
+- The archive **must** contain a single directory (name not important)
+   containing, in turn, the sources. This is the kind of archives
+   automatically generated by GitHub, GitLab, Sourceforge... or through
+   `git archive`.
+- The `alire.toml` manifest must be placed at the top-level with the rest of
+   your sources (inside the same single directory just described), containing
+   all required information except for the `[origin]` table, which will be created
+   by `alr`.
+- This archive **must not** contain the `alire` directory generated by
+   `alr` in working directories. The `alire` directory is designed to be
+    put in your VCS ignore list.
+
+With the source archive already uploaded to the online host where it is going
+to be served (there are no restrictions on this host), you should issue
+
+`alr publish <URL>`
+
+and the publishing process will carry on as in the previous cases, performing
+the checks and providing you with a file to submit to the index, and an upload
+URL to do so.
+
+### Starting with a local source folder
+
+Assisting with the creation an upload of a source archive from local sources, as
+a pre-step to the case just presented, is in our immediate roadmap.
+
+### Support for complex projects whose sources become multiple Alire crates
+
+In case your project does not easily map to a single Alire crate (e.g., because
+you manage multiple project files with different dependencies, or there are
+other reasons to keep the sources together even if they generate several
+crates), at this time you will need to prepare individual online source
+archives (or repositories) and proceed from there.
+
+This is also an important scenario in our roadmap, so stay tuned for more
```

Comment:
I am not convinced that the documentation should make promises roadmap items.

Reply:
OK, dropping these.